utils.py

```diff
@@ -2198,6 +2244,74 @@ def lcm_of_all_minors(G):
     return multi_polylcm(denoms)
 
 
+def poles_and_zeros_of_square_tf_matrix(G):
+    """
+    Determine poles and zeros of a square mimotf matrix, making use of the determinant.
+    This method may fail in special cases. If terms cancel out during calculation of the determinant,
+    not all poles and zeros will be determined.
+
+    Parameters
+    ----------
+    G : mimotf matrix (n x n)
+        The transfer function of the system.
+
+    Returns
+    -------
+    z : array
+        List of zeros.
+    p : array
+        List of poles.
+    cancel : boolean
+             Test whether any terms were cancelled out in determinant calculation.
+
+    Example
+    -------
+    >>> G = mimotf([[tf([1, -1], [1, 2]), tf([4], [1, 2])],
+    ...             [tf([4.5], [1, 2]), tf([2, -2], [1, 2])]])
+    >>> poles_and_zeros_of_square_tf_matrix(G)
+    (array([ 4.]), array([-2.]), False)
+    """
+
+    # Convert mimotf to sympy matrix
+    Gs, s = mimotf2sym(G, deadtime=False)
+
+    # Test cancellation
+    rows, cols = G.shape
+    r = Gs.rank()
+    cancel = False
+    for i in range(rows):
```

Comment:
You only check for possible cancellations using the first row minor expansion. But actually there could be others in the other submatrix calculations. Perhaps we should document this as "possible cancellations" rather than definite cancellations.